code/clue.js

```diff
@@ -9,51 +7,179 @@ const mrGreen = {
+  "Kitchen",
+  "Study",
+  "Library",
+  "Billiard Room",
+  "Lounge",
+  "Ballroom",
+  "Hall",
+  "Spa",
+  "Living Room",
+  "Observatory",
+  "Theater",
+  "Guest House",
+  "Patio"
+]
+
+console.log(rooms[0])
+console.log(weapons[2])
 
-const rooms = []
 
-// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+let mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+}
 
-// This function will be invoked when you click on the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  mystery.killer = randomSelector(suspects);
+  document.getElementById('killerCard').style.background = mystery.killer.color;
+  killerImage.src = mystery.killer.image;
+  document.getElementById('killerName').innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+  document.getElementById('killerAge').innerHTML = `age: ${mystery.killer.age}`;
+  document.getElementById('killerOccupation').innerHTML = `occupation: ${mystery.killer.occupation}`;
+  document.getElementById('killerDescription').innerHTML = mystery.killer.description;
+}
+document.getElementById('killerCard').addEventListener('click', pickKiller)
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
+const pickWeapon = () => {
+  mystery.weapon = randomSelector(weapons)
+  document.getElementById('weaponCard').style.background = mystery.killer.color;
+  document.getElementById('weaponName').innerHTML = mystery.weapon.name;
+  document.getElementById('weaponWeight').innerHTML = `weight: ${mystery.weapon.weight}`;
 }
+document.getElementById('weaponCard').addEventListener('click', pickWeapon)
+
+const pickRoom = () => {
+  mystery.room = randomSelector(rooms)
+  document.getElementById('roomCard').style.background = mystery.killer.color;
+  document.getElementById('roomName').innerHTML = mystery.room;
+}
+document.getElementById('roomCard').addEventListener('click', pickRoom)
+
+const revealMystery = () => {
+  document.getElementById('revealAnswer').innerHTML = `The murder was committed by ${mystery.killer.firstName} ${mystery.killer.lastName}, in the ${mystery.room} with a ${mystery.weapon.name}.`
+}
+document.getElementById('revealAnswer').addEventListener('click', revealMystery)
```

Comment:
You shouldn't be able to reveal the crime until all of the cards are chosen.

Reply:
I have made the change!